c7n/resources/fsx.py

```diff
@@ -0,0 +1,159 @@
+    permissions = ('fsx:TagResource',)
+
+    def process_resource_set(self, resources, tags):
+        client = local_session(self.manager.session_factory).client('fsx')
+        for r in resources:
+            client.tag_resource(ResourceARN=r['ResourceARN'], Tags=tags)
+
+
+@FSx.action_registry.register('remove-tag')
+class UnTagFileSystem(RemoveTag):
+    concurrency = 2
+    batch_size = 5
+    permissions = ('fsx:UntagResource',)
+
+    def process_resource_set(self, resources, tag_keys):
+        client = local_session(self.manager.session_factory).client('fsx')
+        for r in resources:
+            client.untag_resource(ResourceARN=r['ResourceARN'], TagKeys=tag_keys)
+
+
+@FSx.action_registry.register('update')
+class UpdateFileSystem(BaseAction):
+    """
+    Update FSx resource configurations
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+            - name: update-fsx-resource
+              resource: fsx
+              actions:
+                - type: update
+                  WindowsConfiguration:
+                    AutomaticBackupRetentionDays: 1
+                    DailyAutomaticBackupStartTime: '04:30'
+                    WeeklyMaintenanceStartTime: '04:30'
+                  LustreConfiguration:
+                    WeeklyMaintenanceStartTime: '04:30'
+
+    Reference: https://docs.aws.amazon.com/fsx/latest/APIReference/API_UpdateFileSystem.html
+    """
+    permissions = ('fsx:UpdateFileSystem',)
+
+    schema = type_schema(
+        'update',
+        WindowsConfiguration={'type': 'object'},
+        LustreConfiguration={'type': 'object'}
+    )
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('fsx')
+        for r in resources:
+            client.update_file_system(
+                FileSystemId=r['FileSystemId'],
+                WindowsConfiguration=self.data.get('WindowsConfiguration', {}),
+                LustreConfiguration=self.data.get('LustreConfiguration', {})
+            )
+
+
+@FSx.action_registry.register('backup')
+class BackupFileSystem(BaseAction):
+    """
+    Create Backups of File Systems
+
+    :example:
+
+    .. code-block: yaml
+
+        policies:
+            - name: backup-fsx-resource
+              resource: fsx
+              actions:
+                - type: backup
+                  tags:
+                    some-key: some-value
+    """
+    permissions = ('fsx:CreateBackup',)
+
+    schema = type_schema(
+        'backup',
+        **{'tags': {'type': 'object'}}
+    )
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('fsx')
+        tags = [{'Key': k, 'Value': v} for k, v in self.data.get('tags', {}).items()]
+
+        for r in resources:
+            r['Tags'].extend(tags)
+            try:
+                client.create_backup(
+                    FileSystemId=r['FileSystemId'],
+                )
+            except ClientError as e:
```

Comment:
is there a specific error you want to catch? generic error catching is sketchy.

Reply:
yup, forgot to put in the specific error. catching the backup in progress error